django/urls/resolvers.py

```diff
@@ -293,6 +293,11 @@ def _route_to_regex(route, is_endpoint=False):
     return "".join(parts), converters
 
 
+_PATH_ANGLE_BRACKET_COMPONENT_RE = _lazy_re_compile(
+    r"^(?:\/?[^<>\/]*(?:<(?!\/)[^<>\/]*>[^<>\/]*)*)*\/?$"
+)
```

Comment:
This regular expression is way too complicated to understand, verify, and maintain. Can we go with a much easier approach, please? I suggest we find all the angle brackets, and check for unmatched ones with a loop like so:

```python
import re

path = "users/<int:user_id>/comments/<int:comment_id"

brackets = re.findall("[<>]", path)
opened = False
unmatched = False

for b in brackets:
    if b == "<":
        if not opened:
            opened = True
        else:
            unmatched = True
            break
    else:  # >
        if opened:
            opened = False
        else:
            unmatched = True
            break
if opened:
    unmatched = True

if unmatched:
    # errors.append(...)
```

I haven't tested every combination here, please do test it.

Reply:
My first implementation was with using Stack and it will simply match every '<' with a '>' in each segment, otherwise raises a warning. I thought it might be more consistent with the rest of our code, which is using regex.
For the sake of understandability, I can add a comment above that regex explaining it.

Reply:
Please switch to the stack-based approach. Consistency does not trump comprehensibility.

Reply:
I switched. Let me know if it still looks complicated. Plus, I could've used a **counter** instead of a stack, and instead of appending to the stack, make the `counter += 1,` and instead of popping from the stack, make the `counter -=1`. If the `counter < 0`, means it has an extra `'>'`, if the `counter > 0`: means there is an extra `'<'`. 
I wasn't sure if it is more readable than the current stack-based way. If you feel the counter is eloquent enough, I can switch to it.

Reply:
For me, a version with counter is readable enough. Updated.